### DISCUSSION TITLE
Del 2 - presentasjon og logikk

### DISCUSSION THREAD

**Discussion by jorgenhelgheim** [AmroAlnajar/FizzBuzzJazzFuzz#5] [Ideas]
Jeg legger merke til at presentasjonen ikke lenger er likte skilt som i Del 1. Jeg hadde gjerne sett at Console kun ble brukt i program.cs som du gjorde i Del 1. Hva tenker du om det?

**Comment by AmroAlnajar** [comment]
Min tanke var å ha klasser som er så generiske som mulig. Et godt eksempel er bruk av System.Console klassen. Om vi skal bruke System.Console i program.cs så må den brukes hver gang vi skal lage en ny regel og kjøre et nytt spill. 

Istedenfor det, så bruker jeg System.Console et sted (GameBody.RunGame()) for å ha muligheten til å lage et nytt spill med så lite kode som mulig.

Si at jeg skal lage et nytt FizzBuzz spill som kjører på de første 500 nummer.

Klassen GameRunner ser slik ut pr. Idag:

```
using JazzFuzz.RuleSettings;

namespace JazzFuzz.Game;

public class GameRunner : IGameRunner
{
    public void Run()
    {
        new GameBody(new FizzBuzzRuleSettings()).RunGame();
        new GameBody(new JazzFuzzRuleSettings()).RunGame();

        Console.ReadLine();

    }

}
```

Da legger jeg til en ny RuleSetting, og denne linja her:

`new GameBody(new FizzBuzz500RuleSettings()).RunGame();`

Håper dette svarte på spørsmålet :)

**Reply by jorgenhelgheim** [reply]
Men dette betyr at GameBody er bundet opp mot at du må kjøre spille i console. Du får ikke flyttet eller gjenbrukt spillet, uten å endre på GameBody, i en web tilnærming for eksempel. GameBody har endel logikk, så det var derfor jeg tenkte at presentasjon burde utelates fra den klassen, og oppgaven anbefaler at man skal unngå sammenblanding av logikk og presentasjon.

**Reply by AmroAlnajar** [reply]
Godt poeng. Målet var å ha et GameBody klasse som tok for seg all operasjoner som har å gjøre med å kjøre ett spill, men ja, ved å flytte RunGame() metoden ut så er klassen ren logikk. Takk for tips! :)
